examples/mnist/tf_example.py

```diff
@@ -0,0 +1,137 @@
+
+            # Define loss and optimizer
+            y_ = tf.placeholder(tf.int64, [None])
+
+            # The raw formulation of cross-entropy,
+            #
+            #   tf.reduce_mean(-tf.reduce_sum(y_ * tf.log(tf.nn.softmax(y)),
+            #                                 reduction_indices=[1]))
+            #
+            # can be numerically unstable.
+            #
+            # So here we use tf.losses.sparse_softmax_cross_entropy on the raw
+            # outputs of 'y', and then average across the batch.
+            cross_entropy = tf.losses.sparse_softmax_cross_entropy(labels=y_, logits=y)
+            train_step = tf.train.GradientDescentOptimizer(0.5).minimize(cross_entropy)
+
+            train_readout = tf_tensors(train_reader)
+            train_image, train_label = transform(train_readout)
+            train_batch_image, train_batch_label = tf.train.batch(
+                [train_image, train_label], batch_size=batch_size
+            )
+
+            test_readout = tf_tensors(test_reader)
+            test_image, test_label = transform(test_readout)
+            test_batch_image, test_batch_label = tf.train.batch(
+                [test_image, test_label], batch_size=batch_size
+            )
+
+            # Train
+            print('Training model for {0} epochs with batch size {1} and log interval {2}'.format(
+                epochs, batch_size, log_interval
+            ))
+
+            with tf.Session() as sess:
+                sess.run([
+                    tf.local_variables_initializer(),
+                    tf.global_variables_initializer(),
+                ])
+                coord = tf.train.Coordinator()
+                threads = tf.train.start_queue_runners(sess=sess, coord=coord)
+                try:
+                    for i in range(epochs):
+                        if coord.should_stop():
+                            break
+
+                        feed_image_batch, feed_label_batch = sess.run([train_batch_image, train_batch_label])
+                        sess.run(train_step, feed_dict={x: feed_image_batch, y_: feed_label_batch})
+
+                        if (i % log_interval) == 0 or i == (epochs - 1):
+                            correct_prediction = tf.equal(tf.argmax(y, 1), y_)
+                            accuracy = tf.reduce_mean(tf.cast(correct_prediction, tf.float32))
+
+                            feed_image_batch, feed_label_batch = sess.run([test_batch_image, test_batch_label])
+                            print('After training for {0} epochs, the accuracy of the model is: {1:.2f}'.format(
+                                i,
+                                sess.run(accuracy, feed_dict={x: feed_image_batch, y_: feed_label_batch})))
+                finally:
+                    coord.request_stop()
+                    coord.join(threads)
+
+def transform(readout_example):
+    image = tf.cast(tf.reshape(readout_example.image, [784]), tf.float32)
```

Comment:
Consider removing this function all together and just do it inline. It does not do much and the name does help understand its functionality.